### PR TITLE
Add a System Test for the DepolarizedAnalyserTransmission Algorithm

### LINKED ISSUE #37120: Add a system test for DepolarizedAnalyserTransmission
**Describe the outcome that is desired.**
This needs a system test that runs through the whole algorithm once we're certain of what the inputs and outputs should be. 

**Data**

- Empty cell (or MT beam): either `ZOOM38235` or `ZOOM38238` (polariser in)
- Depolarised cell: either `ZOOM38334` (polariser out) or `ZOOM38335` (polariser in - try to use this one for the test)

We have confirmed that these runs are suitable to be made public.

Some pre-processing of the raw data is needed before the workspaces will be ready to pass to the algorithm.

For the empty cell runs:

- Convert to wavelength.
- Extract the spectrum corresponding to monitor 3 into it's own workspace (this is the normalising monitor). This should be spectrum 3.
- Extract the spectrum corresponding to monitor 4 into it's own workspace (this is the monitor between the analyser and main detector that is used for the transmission measurement). This should be spectrum 4.
- Normalise the monitor 4 workspace by monitor 3 (i.e. mon4Ws/mon3Ws).  The normalised workspace is now ready to use in the algorithm.

For the depolarised cell runs (which load as workspace groups with four runs):

- Convert to wavelength.
- Extract the spectrum corresponding to monitor 3 into it's own workspace (this is the normalising monitor). This should be spectrum 3.
- Extract the spectrum corresponding to monitor 4 into it's own workspace (this is the monitor between the analyser and main detector that is used for the transmission measurement). This should be spectrum 4.
- Normalise the monitor 4 workspace by monitor 3 (i.e. mon4Ws/mon3Ws).
- Add the four periods in the normalised monitor 4 workspace and divide by 4 to get a single workspace to pass to the algorithm.

Some examples of the required pre-processing can be seen in Diego's example script.

### PR DESCRIPTION
**Pull request by cailafinn**
#### Summary of work
Add a system test to check the algorithm works consistently across OS' with real data files. 

Fixes #37120 

#### Further detail of work
Also adds the associated data files for running the algorithm properly. The code is set to ignore the error in the fit being quite high. This is intentional. 

### To test:
- Run the algorithm locally using the `./build/systemtest -R DepolarizedAnalyserTransmission` script. 
   - This should download the testing data automatically, but re-running cmake and building the `SystemTestData` target manually may be required. 
- The algorithm should pass without issues. 

*This does not require release notes* because **there are no user-facing changes.**

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: Testing/SystemTests/tests/framework/PolarizationCorrections/DepolarizedAnalyserTransmissionTest.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            compare_alg.setPropertyValue("Workspace1", value_names[0])
+            compare_alg.setPropertyValue("Workspace2", value_names[1])
+            compare_alg.setPropertyValue("Tolerance", str(self.tolerance))
+
+            compare_alg.execute()
+            if compare_alg.getPropertyValue("Result") != "1":
+                print(" Workspaces do not match.")
+                print(self.__class__.__name__)
+                SaveNexus(InputWorkspace=value_names[0], Filename=self.__class__.__name__ + "-mismatch.nxs")
+                return False
```
Comment (jhaigh0) [inline Testing/SystemTests/tests/framework/PolarizationCorrections/DepolarizedAnalyserTransmissionTest.py:59]:
```suggestion
            Load(Filename=reference, OutputWorkspace=reference)
            compare_alg = AlgorithmManager.create("CompareWorkspaces")
            compare_alg.setPropertyValue("Workspace1", result)
            compare_alg.setPropertyValue("Workspace2", reference)
            compare_alg.setPropertyValue("Tolerance", str(self.tolerance))

            compare_alg.execute()
            if compare_alg.getPropertyValue("Result") != "1":
                print(" Workspaces do not match.")
                print(self.__class__.__name__)
                SaveNexus(InputWorkspace=result, Filename=self.__class__.__name__ + "-mismatch.nxs")
                return False
```
Any reason not to just do this? Sorry to be picky

**Review by jhaigh0 — COMMENTED** [review]
Can see the test runs fine in the ci job. One very minor readability thing

**Review by jhaigh0 — COMMENTED** [review]
Would it be easier for the algorithms in the test to use `StoreInADS=False` and then just assign the results to python variables?
If you want to use the ADS, is it easier to just clear the ADS in the `cleanup` method, rather than delete each test by name?

**Comment by cailafinn** [comment]
> Would it be easier for the algorithms in the test to use `StoreInADS=False` and then just assign the results to python variables? If you want to use the ADS, is it easier to just clear the ADS in the `cleanup` method, rather than delete each test by name?

Ah, thanks for the heads up, that's a better solution. I was hesitant to just clear the ADS so it didn't clear other tests. But thinking about it, I don't think they run on the same instance of the framework.

**Review by jhaigh0 — DISMISSED** [review]
Looks good to go, the test failure in the CI is unrelated

**Review by jhaigh0 — APPROVED** [review]

**Pull request merged**